### ISSUE TITLE
OpenWebStart 1.0.0 can not parse our JNLP

### ISSUE THREAD

**Issue by sMOTIVEEd** [karakun/OpenWebStart#103]
Hello developer team,

I just installed OpenWebStart 1.0.0 under Windows to test with out application which runs fine with OracleJDK 8 and included Webstart. It also worked with version 0.3.x of OpenWebStart. But now I get the following exception/starcktrace:

```
--------------- cut here ---------------
Tue Nov 26 16:52:01 CET 2019 [ERROR  ] net.sourceforge.jnlp.AbstractLaunchHandler: netx: Read Error: Could not read or parse the JNLP file. (java.lang.NullPointerException: name can't be null (name can't be null)) net.sourceforge.jnlp.LaunchException: Fatal: Read Error: Could not read or parse the JNLP file. You can try to download this file manually and send it as bug report to IcedTea-Web team.     at net.sourceforge.jnlp.Launcher.fromUrl(Launcher.java:369)     at net.sourceforge.jnlp.Launcher.launch(Launcher.java:228)     at net.sourceforge.jnlp.runtime.Boot.run(Boot.java:296)     at net.sourceforge.jnlp.runtime.Boot.run(Boot.java:78)     at java.security.AccessController.doPrivileged(Native Method)     at net.sourceforge.jnlp.runtime.Boot.main(Boot.java:233)     at net.sourceforge.jnlp.runtime.Boot.main(Boot.java:116) Caused by: java.io.IOException: java.lang.NullPointerException: name can't be null     at net.sourceforge.jnlp.JNLPFile.openURL(JNLPFile.java:386)     at net.sourceforge.jnlp.JNLPFile.<init>(JNLPFile.java:306)     at net.sourceforge.jnlp.JNLPFile.<init>(JNLPFile.java:272)     at net.sourceforge.jnlp.JNLPFile.<init>(JNLPFile.java:257)     at net.sourceforge.jnlp.Launcher.fromUrl(Launcher.java:333)     ... 6 more Caused by: java.lang.NullPointerException: name can't be null     at java.io.FilePermission.init(FilePermission.java:191)     at java.io.FilePermission.<init>(FilePermission.java:277)     at java.lang.SecurityManager.checkRead(SecurityManager.java:888)     at java.io.FileInputStream.<init>(FileInputStream.java:127)     at net.sourceforge.jnlp.JNLPFile.openURL(JNLPFile.java:383)     ... 10 more 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.adoptopenjdk.icedteaweb.resources.cache.LeastRecentlyUsedCache: isUpToDate: http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp - (v: null) = true 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.resources.cache.CacheEntry: http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp - version null: lastModified cache:1574738482000 actual:1574738482000 -> true 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.resources.cache.CacheEntry: http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp - version null: isCached true 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.resources.cache.CacheEntry: isCached: remote size:1980 cached size:1980 -> true 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.sourceforge.jnlp.runtime.JNLPProxySelector: Selected proxies: [DIRECT] 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.sourceforge.jnlp.runtime.JNLPProxySelector: Selecting proxy for: socket://civm03:8080 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.sourceforge.jnlp.runtime.JNLPProxySelector: Selected proxies: [DIRECT] 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.sourceforge.jnlp.runtime.JNLPProxySelector: Selecting proxy for: http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.resources.downloader.BaseResourceDownloader: Downloading location=http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp version=null state=INCOMPLETE from http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.resources.initializer.UnversionedResourceInitializer: Failed to determine best URL for location=http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp version=null state=INCOMPLETE will try all of [http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp] 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.resources.initializer.BaseResourceInitializer: For location=http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp version=null state=INCOMPLETE the server returned 400 code for HEAD request for http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.resources.initializer.UrlProber: URL connection 'http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp' header fields: {null=[HTTP/1.1 400], Connection=[close], Last-Modified=[Tue, 26 Nov 2019 03:21:22 GMT], Content-Length=[1839], Date=[Tue, 26 Nov 2019 15:51:56 GMT], Content-Type=[application/x-java-jnlp-file;charset=ISO-8859-1]} 
 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.http.HttpUtils: Following exception should be harmless, but may help in finding root cause. java.io.IOException: Server returned HTTP response code: 400 for URL: http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp     at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1894)     at sun.net.www.protocol.http.HttpURLConnection.access$200(HttpURLConnection.java:91)     at sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1484)     at sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1482)     at java.security.AccessController.doPrivileged(Native Method)     at java.security.AccessController.doPrivilegedWithCombiner(AccessController.java:782)     at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1481)     at net.adoptopenjdk.icedteaweb.http.CloseableConnection.getInputStream(CloseableConnection.java:58)     at net.adoptopenjdk.icedteaweb.http.HttpUtils.consumeAndCloseConnection(HttpUtils.java:73)     at net.adoptopenjdk.icedteaweb.http.HttpUtils.consumeAndCloseConnectionSilently(HttpUtils.java:60)     at net.adoptopenjdk.icedteaweb.resources.initializer.UrlProber.getUrlResponseCodeWithRedirectionResult(UrlProber.java:71)     at net.adoptopenjdk.icedteaweb.resources.initializer.BaseResourceInitializer.testUrl(BaseResourceInitializer.java:103)     at net.adoptopenjdk.icedteaweb.resources.initializer.BaseResourceInitializer.lambda$null$0(BaseResourceInitializer.java:84)     at java.util.concurrent.FutureTask.run(FutureTask.java:266)     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)     at java.lang.Thread.run(Thread.java:748) 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.sourceforge.jnlp.runtime.JNLPProxySelector: Selected proxies: [DIRECT] 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.sourceforge.jnlp.runtime.JNLPProxySelector: Selecting proxy for: socket://civm03:8080 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.sourceforge.jnlp.runtime.JNLPProxySelector: Selected proxies: [DIRECT] 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.sourceforge.jnlp.runtime.JNLPProxySelector: Selecting proxy for: http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.resources.initializer.UnversionedResourceInitializer: Candidate URLs for location=http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp version=null state=INCOMPLETE: [http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp] 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.adoptopenjdk.icedteaweb.resources.cache.LeastRecentlyUsedCache: isCached: http://civm03:8080/sMOTIVE/archive/sMOTIVE-WebStart-en.jnlp - (v: null) = true 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.resources.cache.CacheEntry: isCached: remote size:1980 cached size:1980 -> true 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.sourceforge.jnlp.Parser: good - your JRE - 1.8.0_201 - match requested JRE - 1.7+ 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.sourceforge.jnlp.JNLPFile: Acceptable vendor tag found, contains: sMOTIVE 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.sourceforge.jnlp.JNLPFile: Acceptable title tag found, contains: sMOTIVE Web Portal 
Tue Nov 26 16:52:01 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesReader: Jars not ready to provide attribute Application-Name 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.sourceforge.jnlp.Parser: Description: sMOTIVE Web Portal 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.sourceforge.jnlp.Parser: Homepage: http://www.smotive.de/ 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.adoptopenjdk.icedteaweb.xmlparser.MalformedXMLParser: Using MalformedXMLParser 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.sourceforge.jnlp.runtime.Boot: JNLP file location: C:\Users\eka\Downloads\sMOTIVE-WebStart(9).jnlp 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.sourceforge.jnlp.runtime.Boot: Proceeding with jnlp 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.sourceforge.jnlp.browser.FirefoxPreferencesParser: Read 384 entries from Firefox's preferences 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.sourceforge.jnlp.browser.FirefoxPreferencesFinder: Found preferences file: 
Tue Nov 26 16:52:01 CET 2019 [INFO  ] net.sourceforge.jnlp.browser.FirefoxPreferencesFinder: Using firefox's profiles file: C:\Users\eka\AppData\Roaming\Mozilla\Firefox\profiles.ini 
Tue Nov 26 16:52:00 CET 2019 [DEBUG  ] net.adoptopenjdk.icedteaweb.client.parts.dialogs.security.SecurityDialogMessageHandler: Starting security dialog thread 
--------------- cut here ---------------
```

I also attached the JNLP-File which I downloaded manually.

Kind regards
Eduard Kappel
[sMOTIVE-WebStart.jnlp.txt](https://github.com/karakun/OpenWebStart/files/3892756/sMOTIVE-WebStart.jnlp.txt)

**Comment by sclassen** [comment]
this could be related to #102
Can you please try the following snapshot and report back if this version fixes the problem for you:
https://cloud.karakun.com/index.php/s/7P43naEYDeDQxgd

**Comment by sMOTIVEEd** [comment]
Version 1.1.0-SNAPSHOT works.

**Comment by hendrikebbers** [comment]
ok, the fix will be in 1.1.0 that we plan to release next week. Will close the issue
